server/s3_object_data_copier.cc

```diff
@@ -122,38 +131,23 @@ void S3ObjectDataCopier::read_data_block_success() {
   }
   // Calculating actial size of data that has just been read
 
-  for (size_t i = 0, n = data_blocks_read.size() - 1; i < n; ++i) {
+  for (size_t i = 0, n = data_blocks_read.size() - 1; i <= n; ++i) {
     const auto motr_block_size = data_blocks_read[i].second;
     assert(motr_block_size == motr_unit_size);
 
     // We can use multiplication, but something may change in the future
     bytes_in_chunk_count += motr_block_size;
   }
   if (bytes_in_chunk_count >= bytes_left_to_read) {
-    s3_log(S3_LOG_ERROR, request_id,
+    s3_log(S3_LOG_DEBUG, request_id,
            "Too many data has been read. Data left - %zu, got - %zu",
            bytes_left_to_read, bytes_in_chunk_count);
-
-    copy_failed = true;
-    set_s3_error("InternalError");
-
-    if (!write_in_progress) {
-      this->on_failure();
-    }
-    return;
-  }
-  bytes_left_to_read -= bytes_in_chunk_count;
-
-  auto& r_last_block_size =
-      data_blocks_read[data_blocks_read.size() - 1].second;
-
-  if (r_last_block_size >= bytes_left_to_read) {
-    r_last_block_size = bytes_left_to_read;
+    get_buffers(data_blocks_read, bytes_in_chunk_count, bytes_left_to_read);
```

Comment:
This function call (get_buffers()) is needed, otherwise the ETAG result of CopyObject api does not match the md5sum stored for the object in object list index table